platform-core-module/src/main/java/com/softicar/platform/core/module/file/smb/ISmbClient.java

```diff
@@ -1,43 +1,80 @@
 package com.softicar.platform.core.module.file.smb;
 
-import java.io.OutputStream;
-
 /**
  * Common interface of all SMB client implementations.
+ * <p>
+ * Provides functionality to access entries (files and folders) on an SMB share.
+ * <p>
+ * {@link CurrentSmbClient} can be used to retrieve an instance.
  *
  * @author Alexander Schmidt
  */
 public interface ISmbClient {
 
 	/**
-	 * Creates a new {@link ISmbFile} from the given URL and
+	 * Creates a new {@link ISmbEntry} handle from the given URL and
 	 * {@link SmbCredentials}.
 	 * <p>
-	 * Note that if the file at the given URL does not exist, it won't get
-	 * physically created until until you create an {@link OutputStream} via
-	 * {@link ISmbFile#createOutputStream()} and write the corresponding data to
-	 * it.
+	 * The entry referenced by the given URL can either be a file or a
+	 * directory.
+	 * <p>
+	 * This method does <b>not</b> physically create a file or directory on the
+	 * respective SMB share.
+	 *
+	 * @param url
+	 *            the URL to the entry on the SMB share (never <i>null</i>)
+	 * @param credentials
+	 *            the credentials to log in to the SMB share (never <i>null</i>)
+	 * @return the {@link ISmbEntry} handle (never <i>null</i>)
+	 */
+	ISmbEntry getEntry(String url, SmbCredentials credentials);
+
+	/**
+	 * Creates a new {@link ISmbFile} handle from the given URL and
+	 * {@link SmbCredentials}.
+	 * <p>
+	 * If the given URL refers to a directory, {@link SmbNoFileException} is
+	 * thrown. For a URL that refers to a directory,
```

Comment:
For an URL

Reply:
Oh, this is the opposite case: URL is pronounced "_Juu_-Arr-Ell", so it's "a" instead of "an".